x/supplier/keeper/msg_server_unstake_supplier_test.go

```diff
@@ -182,11 +194,42 @@ func TestMsgServer_UnstakeSupplier_FailIfCurrentlyUnstaking(t *testing.T) {
 	require.ErrorIs(t, err, types.ErrSupplierIsUnstaking)
 }
 
+func TestMsgServer_UnstakeSupplier_OperatorCanUnstake(t *testing.T) {
+	supplierModuleKeepers, ctx := keepertest.SupplierKeeper(t)
+	srv := keeper.NewMsgServerImpl(*supplierModuleKeepers.Keeper)
+
+	// Generate an address for the supplier
+	ownerAddr := sample.AccAddress()
+	supplierAddr := sample.AccAddress()
+
+	// Stake the supplier
+	initialStake := int64(100)
+	stakeMsg := createStakeMsg(ownerAddr, initialStake)
+	stakeMsg.Address = supplierAddr
+	_, err := srv.StakeSupplier(ctx, stakeMsg)
+	require.NoError(t, err)
+
+	// Initiate the supplier unstaking
+	unstakeMsg := &types.MsgUnstakeSupplier{
+		Signer:  supplierAddr,
+		Address: supplierAddr,
+	}
+	_, err = srv.UnstakeSupplier(ctx, unstakeMsg)
+	require.NoError(t, err)
+
+	// Make sure the supplier entered the unbonding period
+	foundSupplier, isSupplierFound := supplierModuleKeepers.GetSupplier(ctx, supplierAddr)
+	require.True(t, isSupplierFound)
+	require.True(t, foundSupplier.IsUnbonding())
```

Comment:
Do we have a unit test to see where the funds go once unbonding finishes?

Let's add a TODO if not.

Reply:
Good catch!